pkg/repo/callrepo.go

```diff
@@ -93,6 +108,30 @@ func (r *CallsRepository) GetCall(method string) core.Handler {
 	return r.calls[method]
 }
 
+// UpdateCalls refreshes the CallsConfig and rebuilds the handlers for calls accordingly
+// It takes a single parameter that is pointer to the new calls config.
+// The current implementation will completely overwrite the old config with new config.
+// It returns an error to if any while building the handlers, otherwise it returns nil.
+func (r *CallsRepository) UpdateCalls(calls *CallsConfig) {
+	r.mu.Lock()
```

Comment:
@ksysoev can you review this sync implementation? There is one test that I added, it fails here but passes on my machine

Reply:
but here you have the same problem, you need to create new map with all calls first, then just replace it(replace with lock acquired).... the easiest way probably is to adjust createHandler, instead of accepting `*CallsRepository` , accept  `map[string]core.Handler`, and populate calls into separate map

Reply:
to reproduce the issue on you machine,  you  need to run it with race detector.